process/smartContract/processorV2/processV2.go

```diff
@@ -2592,6 +2615,31 @@ func (sc *scProcessor) createSCRForSenderAndRelayer(
 		gasRemaining = 0
 	}
 
+	isRelayedV3 := common.IsValidRelayedTxV3(tx)
+	shouldRefundGasToRelayerSCR = isRelayedV3 && callType != vmData.AsynchronousCall && gasRefund.Cmp(zero) > 0
+	if shouldRefundGasToRelayerSCR {
+		senderForRelayerRefund := tx.GetRcvAddr()
+		if !sc.isSelfShard(tx.GetRcvAddr()) {
```

Comment:
this seems like a big chunk of duplicated code with the If above.

Can you move the creation of refundGasToRelayer into a separate function and make the code smaller?

Reply:
refactored